### ISSUE TITLE
Refresh YOLOX-s Model Zoo package for the Neat 2.1.3 YOLOX decode contract

### ISSUE THREAD

**Issue by oehlke** [sima-neat/sima-cli#172]
Description:

The YOLOX-s package currently distributed through the SiMa Model Zoo 2.1.3 appears to retain a legacy model and postprocessing contract that is incompatible with the public Neat 2.1.3 `BoxDecodeType.YoloX` path.

Current artifact:

- Package: `yolox_s_mpk.tar.gz`
- SHA-256: `5f045de40ba97e0195e2876d1fc02a4faca5c0866f1e89e171cc86794bbd0071`
- Model SDK metadata: 2.1.3
- Detection outputs: three packed 85-channel heads at 80×80, 40×40, and 20×20
- Package decoder type: legacy `yolo`
- Package geometry metadata includes inconsistent/ambiguous 320×320 model dimensions, while the tensor resolutions correspond to the standard 640×640 YOLOX-s topology

Observed behavior:

- The package-default Neat route fails in its legacy detessellate/dequantize path.
- Explicitly selecting `BoxDecodeType.YoloX` fails contract validation because Neat 2.1.3 expects split, head-interleaved tensors:
  - bounding boxes: 4 channels
  - objectness logits: 1 channel
  - class logits: 80 channels
  - repeated at strides 8, 16, and 32
- The model can only currently be benchmarked by terminating after MLA inference, which excludes BoxDecode/NMS.

I verified a compatible implementation using the official Megvii YOLOX-s source and checkpoint:

- Upstream source commit: `6ddff4824372906469a7fae2dc3206c7aa4bbaee`
- Official checkpoint SHA-256: `f55ded7181e1b0c13285c56e7790b8f0e8f8db590fe4edb37f0b7f345c913a30`
- Input: `[1,3,640,640]`
- Outputs: nine raw tensors in Neat `Split3Interleaved` order
- INT8 and BF16 audits: zero unsupported operators
- Model Compiler 2.1.3 compilation: successful
- Neat YOLOX BoxDecode/NMS execution on Modalix: successful

Requested update:

Refresh the Model Zoo YOLOX-s artifact using the current upstream model and generate an MPK whose output and metadata contracts are compatible with Neat 2.1.3.

Acceptance criteria:

- Uses a documented, pinned upstream YOLOX-s source and checkpoint.
- Uses the standard 640×640 YOLOX-s input contract.
- Exposes raw box, objectness-logit, and class-logit tensors in a layout accepted by `BoxDecodeType.YoloX`.
- Records consistent model and source-image geometry.
- Loads through the package-default Neat route without requiring `mla_only`.
- Successfully runs inference through YOLOX BoxDecode and NMS on Modalix.
- Includes a smoke test using the public `pyneat.Model` API.
- Replaces or clearly deprecates the stale Model Zoo download.
- Documents the updated artifact checksum and compatibility version.

**Comment by oehlke** [comment]
Thanks. I reproduced both paths against the unchanged Model Zoo artifact on
the paired Modalix DevKit and captured the complete logs.

Artifact:

```text
/workspace/modelzoo/2.1.3/yolox_s_mpk.tar.gz
SHA-256: 5f045de40ba97e0195e2876d1fc02a4faca5c0866f1e89e171cc86794bbd0071
Python: 3.11.2
Public pyneat module: /media/nvme/pyneat/lib/python3.11/site-packages/pyneat/__init__.py
```

### 1. Package-default route

Public API call:

```python
model = pyneat.Model(model_path)
model.benchmark(1)
```

The model constructs and the graph reports ready, but processing the benchmark
input fails as follows:

```text
exception_type=NeatError
exception=[runtime.element_failed] A pipeline stage stopped while processing data.

Stage: detessdequant_1

How to fix:
- Correct the configuration of the reported stage and its upstream input.
- Restart the pipeline after correcting the first upstream failure.

Diagnostic ID: gstreamer.unclassified_element_failure
```

### 2. Explicit public YOLOX route

Public API configuration:

```python
options = pyneat.ModelOptions()
options.decode_type = pyneat.BoxDecodeType.YoloX
options.decode_type_option = pyneat.BoxDecodeTypeOption.Split3Interleaved
options.num_classes = 80
options.score_threshold = 0.25
options.nms_iou_threshold = 0.45
options.top_k = 100
model = pyneat.Model(model_path, options)
model.benchmark(1)
```

The graph again reports ready. The decoder then prints:

```text
Width not the same in compatible tensors: 80,1
[boxdecode-faildump] rc=-493 stage=boxdecode_1 decode=yolox mode=INT8_TESS_QUANT num_classes=80 model=320x320 original=320x320 num_in_tensor=3 ...
```

Relevant fields from the faildump are:

```text
num_in_tensor: 3
input_depth: [80,40,20]
input_height: [85,85,85]
input_width: [80,40,20]
slice_depth: [80,40,20]
slice_height: [1,1,1]
slice_width: [80,40,20]
```

The public exception is:

```text
exception_type=NeatError
exception=[io.parse] The reported pipeline stage configuration is invalid.

Stage: boxdecode_1

How to fix:
- Correct the configuration for the reported stage.
- Validate the configuration syntax and values before starting the pipeline.

Diagnostic ID: gstreamer.configuration_invalid
```

This confirms the observed failure mode without asserting which release or
artifact-generation component introduced it: the explicit YOLOX path receives
three packed 85-channel heads rather than the nine split box/objectness/class
tensors requested by `Split3Interleaved`, and `boxdecode_1` rejects the derived
configuration.

As controls, the same artifact runs when the public model route is explicitly
terminated at MLA output (`ModelOptions.inference_terminal.mla_only=True`), and
a separately compiled nine-output YOLOX-s package successfully runs through the
same public `YoloX`/`Split3Interleaved` BoxDecode/NMS path.

I can attach the two complete captured logs if preferred.
